instruments/KEIT196.py

```diff
@@ -0,0 +1,102 @@
+from abstract_instrument import abstract_instrument
+import socket
+
+#==============================================================================
+
+ALL_VAL_TYPE = ['DCV', 'ACV', 'DCI', 'ACI', 'RES']  #, 'PERIOD']
+ALL_CHANNELS = ['1'] #, '2']
+
+ADDRESS = "10.1.28.59"
+ADDITIONAL_ADDRESS = "7"
+CONF_VAL_TYPE = ['F0DX', 'F1DX', 'F3DX', 'F4DX', 'F2DX'] #, 'CONF:PERIOD']
+
+#==============================================================================
+
+class KEIT196(abstract_instrument):
+	def __init__(self, channels, vtypes, address, additional_address):
+		self.address = address
+		self.port = 1234
+		self.gpib_addr = additional_address
+		self.channels = channels
+		self.vtypes = vtypes
+
+	def model(self):
+		#self.send("*IDN?")
+		#return self.read()
+		return "KEIT196"
+
+	def connect(self):
+		print('Connecting to device @%s:%s GPIB:%s...' %(self.address, self.port, self.gpib_addr))
+		self.sock = socket.socket(socket.AF_INET,
+							 socket.SOCK_STREAM,
+							 socket.IPPROTO_TCP)
+		self.sock.settimeout(10.0)	# Don't hang around forever
+		self.sock.connect((self.address, self.port))
+		self.init_prologix()
+		print('  --> Ok')
+		print(self.model())
+		self.configure()
+
+	def configure(self):
+		self.send('*RST')
+		#self.send('F0DX')
+		for ch in self.channels:
+			self.send(CONF_VAL_TYPE[ALL_VAL_TYPE.index(self.vtypes[self.channels.index(ch)])])
+		self.send('YX')
+		self.send('G1DX')
+		self.send('R0DX')
+		self.send('W100') #delay in milliseconds 0ms -> 6000ms
+
+	def getValue(self):
+		#self.send('FETC?')
+		#self.send('READ:FREQ?')
+		return self.read()
+
+	def read(self):
+		self.send("++read eoi")
+		ans = ''
+		nb_data_list = []
+		nb_data = ''
+		try:
+			while ans != '\n':
+				ans = self.sock.recv(1)
+				nb_data_list.append(ans) # Return the number of data
+			list_size = len(nb_data_list)
+			for j in range (0, list_size):
+				nb_data = nb_data+nb_data_list[j]
+			return nb_data
+		except socket.timeout:
+			print "Socket timeout error when reading."
+			raise
+		self.send('*RST')
+		self.send('YX')
+		self.send('G1DX')
+		self.send('R0DX')
+		self.send('W100')
```

Comment:
Are you sure you have to reset at each acquisition ? Should be a heavy process on the instrument side.